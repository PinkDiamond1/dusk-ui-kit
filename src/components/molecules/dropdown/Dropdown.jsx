/* @flow */
/** @jsx jsx */
import { jsx } from '@emotion/core'
import * as React from 'react'
import uuid from 'react-uuid'
import BootstrapDropdown from 'react-bootstrap/Dropdown'

import DropdownItem from '../../atoms/dropdown-item/DropdownItem'

type Props = {
  id: string,
  as: React.ElementType,
  children: React.Node,
  className: string,
  onToggle: Function,
  onSelect: Function,
  variant: 'primary' | 'secondary' | 'info' | 'success' | 'warning' | 'danger',
}

const Dropdown = (props: Props): React.Element<*> => {
  const { children, className, onToggle, onSelect, variant, as, id } = props
  const updatedClassName = `__duk-dropdown ${className}`
  return (
    <BootstrapDropdown as={as} className={updatedClassName} onToggle={onToggle} onSelect={onSelect}>
      <BootstrapDropdown.Toggle variant={variant} id={id}>
        Dropdown
      </BootstrapDropdown.Toggle>
      <BootstrapDropdown.Menu>{children}</BootstrapDropdown.Menu>
    </BootstrapDropdown>
  )
}

Dropdown.defaultProps = {
  id: uuid(),
  as: 'div',
  className: '',
  onToggle: () => {},
  onSelect: () => {},
  variant: 'primary',
}

Dropdown.Item = DropdownItem

export default Dropdown
