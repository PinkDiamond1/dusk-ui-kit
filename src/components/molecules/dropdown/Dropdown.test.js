import React from 'react'
import renderer from 'react-test-renderer'
// import jest from 'jest'

import Dropdown from './Dropdown'

// jest.mock('uuid', () => ({
//   v4: jest.fn(() => 1),
// }))

// it('renders Dropdown', () => {
//   const tree = renderer.create(<Dropdown>Smoke Test</Dropdown>).toJSON()
//   expect(tree).toMatchSnapshot()
// })

it('renders Dropdown with any uuid number', () => {
  const tree = renderer.create(<Dropdown>Smoke Test</Dropdown>).toJSON()
  expect(tree).toMatchSnapshot()
})
