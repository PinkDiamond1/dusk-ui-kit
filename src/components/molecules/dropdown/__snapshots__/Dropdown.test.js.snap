// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Dropdown 1`] = `
<div
  className="__duk-dropdown  dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-primary"
    disabled={false}
    id="test"
    onClick={[Function]}
    type="button"
  />
</div>
`;
